sway/criteria.c

```diff
@@ -376,31 +376,31 @@ static char *get_focused_prop(enum criteria_token token, bool *autofail) {
 	struct sway_container *focus = seat_get_focused_container(seat);
 
 	struct sway_view *view = focus ? focus->view : NULL;
-	const char *value = NULL;
+	char *value = NULL;
 
 	switch (token) {
 	case T_APP_ID:
 		*autofail = true;
 		if (view) {
-			value = view_get_app_id(view);
+			value = strdup(view_get_app_id(view));
```

Comment:
If `view_get_app_id` can return NULL, it will segfault.

Reply:
Indeed! This one is a bit akward to fix without breaking the pattern of other cases. Please let me know what you think of the new solution.